ledger/src/blockstore.rs

```diff
@@ -1239,15 +1242,15 @@ impl Blockstore {
             let maybe_shred = self.get_coding_shred(slot, coding_index);
             if let Ok(Some(shred_data)) = maybe_shred {
                 let potential_shred = Shred::new_from_serialized_shred(shred_data).unwrap();
-                if Self::erasure_mismatch(&potential_shred, shred) {
-                    return Some(potential_shred.payload);
+                if shred.erasure_mismatch(&potential_shred).unwrap_or_default() {
```

Comment:
might just be able to hard unwrap() here, we should never enter this code path with non coding shreds

Reply:
changed to `unwrap`